test/collection_serializer_test.rb

```diff
@@ -23,6 +33,11 @@ def build_named_collection(*resource)
         resource
       end
 
+      def build_named_collection_with_matching_serializer(*resource)
+        resource.define_singleton_method(:name) { 'SerializedResource' }
```

Comment:
When I first came across this weirdness, it was actually while testing, and on our API I wasn't seeing the issue. I just did some more poking around and found that it might actually have something to do with the `#none` method.

In this case, `tags` is the expected json_key as defined in the `ResourceTagSerializer`.

```
[1] pry(main)> ActiveModel::Serializer::CollectionSerializer.new(ResourceTag.none, {}).json_key
=> "resource_tags"
[2] pry(main)> ActiveModel::Serializer::CollectionSerializer.new(ResourceTag.last(3), {}).json_key
=> "tags"
```